### PR TITLE
fix: insert instead of update new state with updated revision during migration

### PR DESCRIPTION
**Pull request by sebastian-alfers**
Reopening of #554, follow up to https://github.com/akka/akka-persistence-r2dbc/pull/560

Open tasks:
- [x] look into adding tests for additional column and change handler

### CODE REVIEW

File: core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresDurableStateDao.scala [hunk 1]
```diff
@@ -93,6 +93,8 @@ private[r2dbc] class PostgresDurableStateDao(executorProvider: R2dbcExecutorProv
 
   private val sqlCache = Sql.Cache(settings.numberOfDataPartitions > 1)
 
+  protected def shouldInsert: (SerializedStateRow) => Future[Boolean] = state => Future.successful(state.revision == 1)
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresDurableStateDao.scala:96]: can't this be an ordinary `def`?

```
protected def shouldInsert(state: SerializedStateRow): Future[Boolean] = {
  Future.successful(state.revision == 1)
}
```

File: migration-tests/src/test/scala/akka/persistence/r2dbc/migration/MigrationTestColumn.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import akka.persistence.r2dbc.state.scaladsl.AdditionalColumn
+
+class MigrationTestColumn extends AdditionalColumn[Any, String] {
+
+  override val columnName: String = "test_column"
+
+  override def bind(upsert: AdditionalColumn.Upsert[Any]): AdditionalColumn.Binding[String] = {
+    if (upsert.value.asInstanceOf[String] == "s-column") {
+      println(s"??? ${upsert.value.asInstanceOf[String]}")
```
Comment (patriknw) [inline migration-tests/src/test/scala/akka/persistence/r2dbc/migration/MigrationTestColumn.scala:15]: remove?

**Review by patriknw — COMMENTED** [review]
looking good, just something small...

File: core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresDurableStateDao.scala (hunk 1 above)
Comment (sebastian-alfers) [reply core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresDurableStateDao.scala:96]: Yes, of course!

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**